cinema/views.py

```diff
@@ -19,26 +28,52 @@
     OrderSerializer,
     OrderListSerializer,
 )
+from user.permissions import IsAdminOrIfAuthenticatedReadOnly
 
 
-class GenreViewSet(viewsets.ModelViewSet):
+class GenreViewSet(
+    viewsets.GenericViewSet,
+    mixins.ListModelMixin,
+    mixins.CreateModelMixin
+):
     queryset = Genre.objects.all()
     serializer_class = GenreSerializer
+    authentication_classes = (TokenAuthentication,)
+    permission_classes = (IsAdminOrIfAuthenticatedReadOnly,)
```

Comment:
One tip. Read about REST_FRAMEWORK constant
https://www.django-rest-framework.org/#example
It'll help you to make default authentication class, permissions and so on so you don't need to copy-paste this code everywhere. And it case you need to override somewhere it, you'll be able to do it